### PR TITLE
Angular: Fix void element selectors

### PR DESCRIPTION
**Pull request by stefan-schweiger**
Issue: #15494

## What I did
Check during `buildTemplate` if the element is a void element and use a different replacement strategy for it.

## How to test

- Is this testable with Jest or Chromatic screenshots?
I've added two basic jest tests for this

### CODE REVIEW

**Comment by manuelmeister** [comment]
@ThibaudAV I think, this PR/issue would be appreciated.

File: app/angular/src/client/preview/angular-beta/ComputesTemplateFromComponent.ts [hunk 1]
```diff
@@ -153,11 +173,16 @@ const buildTemplate = (
     [/#([\w-]+)/, ` id="$1"`],
     [/((\.[\w-]+)+)/, (_, c) => ` class="${c.split`.`.join` `.trim()}"`],
     [/(\[.+?])/g, (_, a) => ` ${a.slice(1, -1)}`],
```
Comment (ThibaudAV) [inline app/angular/src/client/preview/angular-beta/ComputesTemplateFromComponent.ts:175]: How about adding this ? it has the same effect as your proposal but keeps the concept of the previous code 

```suggestion
[
      /([\S]+)(.*)/,
      (template, elementSelector) => {
        return voidElements.some((element) => elementSelector === element)
          ? template.replace(/([\S]+)(.*)/, `<$1$2${inputs}${outputs} />`)
          : template.replace(/([\S]+)(.*)/, `<$1$2${inputs}${outputs}>${innerTemplate}</$1>`);
      },
    ],
    ```

**Comment by ThibaudAV** [comment]
good catch and thanks for the change request ❤️

**Comment by stefan-schweiger** [comment]
@ThibaudAV what would be needed to merge this PR?

**Comment by ThibaudAV** [comment]
@stefan-schweiger can you look at my proposal here? https://github.com/storybookjs/storybook/pull/15495#discussion_r764094330

**Comment by stefan-schweiger** [comment]
@ThibaudAV looks good to me, I've just accepted the changes 😉

**Comment by stefan-schweiger** [comment]
@shilman any chance to get this into 6.5?

**Comment by shilman** [comment]
@stefan-schweiger Happy to merge this once CI is passing. Can you please take a look at the failing unit tests?

**Comment by stefan-schweiger** [comment]
@shilman ok I think now everything should be back in order again. (At least as far as I can tell the broken e2e tests aren't my fault)

Would it now be possible to merge this?

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**